Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerMovement : MonoBehaviour
+{
+    private Rigidbody _rb;
+
+    [Header("Movement")]
+    [SerializeField] private float _moveSpeed = 10;
+    public float MoveSpeed { get { return _moveSpeed; } }
+    [SerializeField] private float _maxForce = 1;
+    [SerializeField] private float _gravity = 9.8f;
+    [SerializeField] private float _gravityWhenGrounded = 2;
+    [SerializeField] private float _groundDrag = 1;
+
+    [Header("Ground check")]
+    [SerializeField] private Transform _groundCheckPoint;
+    [SerializeField] private float _groundCheckRadius = 1;
+    [SerializeField] private LayerMask _groundLayer;
+    [SerializeField] private float _hitAngle = 50;
+    private bool _grounded = false;
+    public bool Grounded { get { return _grounded; } }
+
+    public bool UseGravity { get; set; } = true;
+
+    [Header("Jump")]
+    [SerializeField] private float _jumpForce = 10;
+    private bool _jumping = false;
+
+    private Vector2 _movementInput = Vector2.zero;
+    public Vector2 MovementInput { get { return _movementInput; } }
+    private Vector3 _movementDir = Vector3.zero;
+    public Vector3 MovementDir { get { return _movementDir; } }
+    private Vector3 _hitNormal;
+
+    private Camera _cam;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        _cam = Camera.main;
+        _rb = GetComponent<Rigidbody>();
+        _rb.useGravity = false;
+    }
+
+    private void Update()
+    {
+        GroundCheck();
+
+        if (_rb.velocity.y < 0)
+            _jumping = false;
+    }
+
+    void FixedUpdate()
+    {
+        ApplyGravity();
+        UpdateMovement();
+    }
+
+    private void ApplyGravity()
+    {
+        if (!UseGravity)
+            return;
+
+        if (_grounded && !_jumping)
+        {
+            _rb.AddForce(-_hitNormal * _rb.mass * _gravityWhenGrounded);
+            return;
+        }
+
+        _rb.AddForce(Vector3.down * _rb.mass * _gravity);
```

Comment:
```suggestion
            _rb.AddForce(-_hitNormal * (_rb.mass * _gravityWhenGrounded));
            return;
        }

        _rb.AddForce(Vector3.down * (_rb.mass * _gravity));
```

Order of multiplication operations is inefficient.
Vector3 * float * float == bad, because vector times float, then again vector times other float. Vector3 are 3 floats.
Vector3 * (float * float) == good, because making the 2 floats into 1.